solr/packaging/build.gradle

```diff
@@ -238,6 +238,7 @@ task integrationTests(type: BatsTask) {
   def integrationTestOutput = "$buildDir/test-output"
   def solrHome = "$integrationTestOutput/solr-home"
   def solrTestFailuresDir = "$integrationTestOutput/failure-snapshots"
+  def solrPort = new Random().nextInt(10000, 50000)
```

Comment:
Future improvement idea: check that the ports chosen aren't actually in use by anything before the test(s). I like this improvement, but there is no guarantee that a run won't conflict with an already allocated port on the machine.

Reply:
True. Do you know a good Java library / API to check it (other than trying to listen to a socket and catch the exception)?

Reply:
not really - even https://stackoverflow.com/questions/2675362/how-to-find-an-available-port which is used elsewhere in Solr doesn't guarantee that port won't be used immediately after. Its an idea though

Reply:
Why do we need three consecutive ports? Get three random ports from :0

Reply:
Yep, will test binding to zero now.

Reply:
That worked, except it picked 648xx once, and then solr complained that the embedded ZK on +1000 could not start. So I now use a while loop until we get an acceptable port. Then I still add +1 and +2 to the other nodes, since else we'd need to factor out the port finder as a function and I don't know how that looks like in gradle, and you'd still not be guaranteed unique ports, since it could suggest the same port twice :) Think this is about 100x better than what we had...